component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java

```diff
@@ -1129,4 +1130,27 @@ public static Map<String, String> getClaimProperties(String tenantDomain, String
         }
         return Collections.emptyMap();
     }
+
+    /**
+     * Returns the encrypted claim value if EnableEncryption property is available for the claim. Otherwise, return the
+     * plain text.
+     *
+     * @param claimURI     Claim URI.
+     * @param claimValue   Claim value.
+     * @param tenantDomain Tenant domain.
+     * @return Encrypted or plain claim value based on the EnableEncryption claim property.
+     * @throws TOTPAuthenticatorException When errors occurs while encrypting the claim value.
+     */
+    public static String getProcessedClaimValue(String claimURI, String claimValue, String tenantDomain) {
+
+        Map<String, String> claimProperties = getClaimProperties(tenantDomain, claimURI);
```

Comment:
Is this null safe ? can claimProperties be null ?

Reply:
No claimProperties cannot be null. getClaimProperties will return an empty map if no properties are available. [https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/159/files#diff-69aacd3727454234e50c60bafac6f3fa57cf7154db56021ced438a640608cde1R1121](https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/159/files#diff-69aacd3727454234e50c60bafac6f3fa57cf7154db56021ced438a640608cde1R1121)